content/05.main-text.md

```diff
@@ -278,41 +278,68 @@ The second uses a Cypher query, prepared by [`hetnetpy.neo4j.construct_dwpc_quer
+#### Details of matrix DWPC implementation
 
-Mention approximations. Also note independent approximation work at https://github.com/mmayers12/hetnet_ml
+DWPC computation requires us to remove all duplicate nodes from paths.
+We used three repeat patterns as the building blocks for DWPC computation: short repeats (AAA), nested repeats (BAAB), and overlapping repeats (BABA).
+Let $\mathrm{D}(\mathit{XwXyZ})$ denote the DWPC matrix for metapath _XwXyZ_.
+Under this notation, $\mathrm{D}(\mathit{XyZ})$ is the weighted (bi)adjacency matrix for metaedge _XyZ_.
+Additionally, let $\mathrm{diag}(A)$ represent a diagonal matrix whose entries are the diagonal elements of $A$.
 
-Discuss caching strategies, sequential versus recursive
+For the case of short (< 4) repeats for a single metanode, _XaXbX_ (eg. _GiGdG_), we simply subtract the main diagonal.
 
-Runtime comparison to show the speedup.
-Rephetio computed a portion of in XX time
+$$\mathrm{D}(\mathit{XaXbX}) = \mathrm{D}(\mathit{XaX}) \mathrm{D}(\mathit{XbX}) - \mathrm{diag}(\mathrm{D}(\mathit{XaX}) \mathrm{D}(\mathit{XbX}))$$
 
+Nested repeats _XaYbYcX_ (eg. _CtDrDtC_), are treated recursively, with both inner (YY) and outer (XX) repeats treated as separate short repeats.
 
-From [@vagelos-2017]:
+$$
+\mathrm{D}(\mathit{XaYbYcX}) = \mathrm{D}(\mathit{XaY}) \mathrm{D}(\mathit{YbY}) \mathrm{D}(\mathit{YcX}) - \mathrm{diag}(\mathrm{D}(\mathit{XaY}) (\mathrm{D}(\mathit{YbY}) \mathrm{D}(\mathit{YcX}))
+$$
+
+Overlapping repeats _XaYbXcY_ (eg. _CtDtCtD_) require several corrections ($\odot$ denotes the Hadamard product).
+
+\begin{align}
+\mathrm{D}(\mathit{XaYbXcY}) =\ &\mathrm{D}(\mathit{XaY})\  \mathrm{D}(\mathit{YbX})\  \mathrm{D}(\mathit{XcY}) \\
+&- \mathrm{diag}(\mathrm{D}(\mathit{XaY})\  \mathrm{D}(\mathit{YbX}))\  \mathrm{D}(\mathit{XcY}) \\
+&- \mathrm{D}(\mathit{XaY})\  \mathrm{diag}(\mathrm{D}(\mathit{YbX})\  \mathrm{D}(\mathit{XcY})) \\
+&+ \mathrm{D}(\mathit{XaY})\  \odot \mathrm{D}(\mathit{YbX})^T\  \odot \mathrm{D}(\mathit{XcY})
+\end{align}
+
+Most paths of length six---and many even longer paths---can be represented hierarchically using these patterns.
+For example, a long metapath pattern of the form CBABACXYZ can be segmented as (C(BABA)C)XYZ using patterns for short and overlapping repeats and can be computed using the tools we developed.
+In addition to these matrix routines---which advantageously count rather than enumerate paths---we implemented a general matrix method for any metapath type.
+The general method is important for patterns such as long ($\geq$ 4) repeats, or complex repeat patterns (eg. of the form ABCABC), but it requires path enumeration and is therefore slower.
+As an alternative approach for complex paths, we developed an approximate DWPC method that corrects repeats in disjoint simple patterns but only corrects the first repeat in complex patterns (eg. $\geq$ length four repeat).
```

Comment:
```suggestion
As an alternative approach for complex paths, we developed an approximate DWPC method that corrects repeats in disjoint simple patterns but only corrects the first repeat in complex patterns (e.g. ≥ length four repeat).
```

since there's more overhead to rendering equations, best to avoid math mode when a unicode symbol exists.